pkg/ipam/node.go

```diff
@@ -469,7 +469,7 @@ func (n *Node) ResourceCopy() *v2.CiliumNode {
 // attaches it to the instance as specified by the CiliumNode. neededAddresses
 // of secondary IPs are assigned to the interface up to the maximum number of
 // addresses as allowed by the instance.
-func (n *Node) createInterface(ctx context.Context, a *AllocationAction) error {
+func (n *Node) createInterface(ctx context.Context, a *AllocationAction) (bool, error) {
```

Comment:
nit: Could you give this boolean return value a name, e.g. `instanceMutated`? It's not clear without context.

Thanks!